airbyte-ci/connectors/pipelines/pipelines/pipelines/metadata.py

```diff
@@ -2,112 +2,100 @@
+from pipelines.steps.poetry_run_step import PoetryRunStep
+from pipelines.steps.simple_ci_step import SimpleCIStep, MountPath
+from pipelines.utils import DAGGER_CONFIG, get_secret_host_variable
 
 METADATA_DIR = "airbyte-ci/connectors/metadata_service"
 METADATA_LIB_MODULE_PATH = "lib"
+METADATA_SERVICE_TOOL_MODULE_PATH = Path(f"{METADATA_DIR}/{METADATA_LIB_MODULE_PATH}")
 METADATA_ORCHESTRATOR_MODULE_PATH = "orchestrator"
 
-# HELPERS
-
-
-def get_metadata_file_from_path(context: PipelineContext, metadata_path: Path) -> dagger.File:
-    if metadata_path.is_file() and metadata_path.name != METADATA_FILE_NAME:
-        raise ValueError(f"The metadata file name is not {METADATA_FILE_NAME}, it is {metadata_path.name} .")
-    if metadata_path.is_dir():
-        metadata_path = metadata_path / METADATA_FILE_NAME
-    if not metadata_path.exists():
-        raise FileNotFoundError(f"{str(metadata_path)} does not exist.")
-    return context.get_repo_dir(str(metadata_path.parent), include=[METADATA_FILE_NAME]).file(METADATA_FILE_NAME)
-
-
-def get_metadata_icon_file_from_path(context: PipelineContext, metadata_icon_path: Path) -> dagger.File:
-    return context.get_repo_dir(str(metadata_icon_path.parent), include=[METADATA_ICON_FILE_NAME]).file(METADATA_ICON_FILE_NAME)
-
 
 # STEPS
 
 
-class PoetryRun(Step):
-    def __init__(self, context: PipelineContext, title: str, parent_dir_path: str, module_path: str):
-        self.title = title
-        super().__init__(context)
-        self.parent_dir = self.context.get_repo_dir(parent_dir_path)
-        self.module_path = module_path
-        self.poetry_run_container = with_poetry_module(self.context, self.parent_dir, self.module_path).with_entrypoint(["poetry", "run"])
-
-    async def _run(self, poetry_run_args: list) -> StepResult:
-        poetry_run_exec = self.poetry_run_container.with_exec(poetry_run_args)
-        return await self.get_step_result(poetry_run_exec)
-
-
-class MetadataValidation(PoetryRun):
-    def __init__(self, context: PipelineContext, metadata_path: Path):
-        title = f"Validate {metadata_path}"
-        super().__init__(context, title, METADATA_DIR, METADATA_LIB_MODULE_PATH)
-        self.poetry_run_container = self.poetry_run_container.with_mounted_file(
-            METADATA_FILE_NAME, get_metadata_file_from_path(context, metadata_path)
+class MetadataValidation(SimpleCIStep):
+    def __init__(self, context: ConnectorContext):
+        super().__init__(
+            title=f"Validate metadata for {context.connector.technical_name}",
+            context=context,
+            paths_to_mount=[
+                MountPath(context.connector.metadata_file_path),
+                MountPath(context.connector.documentation_file_path),
+                MountPath(context.connector.inapp_documentation_file_path, optional=True),
+                MountPath(context.connector.icon_path, optional=True),
+            ],
```

Comment:
⚠️ I think we're missing the migration file path here!

But i'd definitely have noticed the spot to add it more quickly the first time if it were formatted like this!

Reply:
migration file path needs to be in the list of mounted paths, but not in the documentation to upload (yet, at least) because it's used in the pre-upload validation, but never shown directly in the product. (Again, yet... I'm having thoughts about that modal...)

Reply:
At least I'm pretty sure - though I can't find the line in the deleted code so maybe i'm thinking about something different

Reply:
https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/connector_ops/connector_ops/qa_checks.py#L14

The check that the migration guide is properly formatted is part of QA checks. Not metadata
https://github.com/airbytehq/airbyte/blob/bnchrch/ci/metadata-doc-step/airbyte-ci/connectors/connector_ops/connector_ops/qa_checks.py#L14

Though this is confusing. Theres definitely an argument to be made that this should be the same code location

Reply:
Aha. Was almost certain I was missing something... Agreed it is confusing, but I don't think that's a problem for this pr.